src/coreclr/src/jit/earlyprop.cpp

```diff
@@ -83,17 +85,29 @@ GenTree* Compiler::getArrayLengthFromAllocation(GenTree* tree DEBUGARG(BasicBloc
                 call->gtCallMethHnd == eeFindHelper(CORINFO_HELP_NEWARR_1_VC) ||
                 call->gtCallMethHnd == eeFindHelper(CORINFO_HELP_NEWARR_1_ALIGN8))
             {
-#ifdef DEBUG
-                optCheckFlagsAreSet(OMF_HAS_NEWARRAY, "OMF_HAS_NEWARRAY", BBF_HAS_NEWARRAY, "BBF_HAS_NEWARRAY", tree,
-                                    block);
-#endif
                 // This is an array allocation site. Grab the array length node.
-                return gtArgEntryByArgNum(call, 1)->GetNode();
+                rtn = gtArgEntryByArgNum(call, 1)->GetNode();
+            }
+            else if (call->gtCallMethHnd == eeFindHelper(CORINFO_HELP_READYTORUN_NEWARR_1))
+            {
+#if defined(TARGET_WINDOWS) && defined(TARGET_ARM)
```

Comment:
Why is this code specific to `defined(TARGET_WINDOWS) && defined(TARGET_ARM)`? Where is the platform-specific code creating this argument in a different location?

Reply:
It's this bit in morph that's prepending handles: https://github.com/dotnet/runtime/blob/667e71dcd6ddeaa0317f68454afd3da8f9368781/src/coreclr/src/jit/morph.cpp#L2746-L2771

It seems this only happens on windows arm, despite the JIT side code indicating that it can happen elsewhere. I've changed the logic to always grab the last argument on arm arch just so we aren't coupled with with the runtime logic dictating this, though I'm not entirely convinced that this is the right play.